crypto/aes/aes_ige.c

```diff
@@ -46,7 +46,7 @@ void AES_ige_encrypt(const unsigned char *in, unsigned char *out,
     size_t n;
     size_t len = length;
 
-    if (length == 0)
+    if (length / AES_BLOCK_SIZE == 0)
```

Comment:
we have this `OPENSSL_assert((length % AES_BLOCK_SIZE) == 0)` a few lines later,
since this is a void function there is no way of returning a result, so
you could instead change the assert below to `if (!OPENSSL_assert(...)) return;`

Reply:
In that case, `ossl_assert` is preferred

Reply:
AES_ige_encrypt is an API so i didnt want to abort() applications which do this wrong and just do nothing instead (doing nothing is the current behavior, just that it leads to overlapping memcpy).

Reply:
`ossl_assert` only aborts when building in debug mode, otherwise it returns the result of the assertion condition.

Reply:
> AES_ige_encrypt is an API so i didnt want to abort() applications which do this wrong and just do nothing instead (doing nothing is the current behavior, just that it leads to overlapping memcpy).

Are you sure?  OPENSSL_assert calls OPENSSL_die, unconditionally, so
the current behaviour is abort, or am I missing something?

Reply:
> OPENSSL_assert calls OPENSSL_die, unconditionally
Yes that was exactly my point: the function did not do OPENSSL_assert((length / AES_BLOCK_SIZE) == 0) and i didnt want to add such an assertion because of the unconditional abort.

Reply:
i no changed it to ossl_assert, so only DEBUG builds will die, indicating a missuse of the function, while production builds will continue to do nothing.